src/Pos/Explorer/Txp/Init.hs

```diff
@@ -0,0 +1,46 @@
+-- | Routines for initialization of Explorer's DB.
+
+module Pos.Explorer.Txp.Init
+       ( initializeGenesisBalances
+       ) where
+
+import           Universum
+
+import qualified Data.Map.Strict             as M
+
+import           Pos.Binary.Class            (encodeStrict)
+import           Pos.Context.Class           (WithNodeContext)
+import           Pos.Context.Functions       (genesisUtxoM)
+import           Pos.DB.Class                (MonadDB)
+import           Pos.DB.GState.Common        (gsGetBi, gsPutBi)
+import           Pos.Explorer.Txp.Toil.Class (MonadTxExtra (..))
+import           Pos.Txp.Core                (TxOut (..), TxOutAux (..))
+import           Pos.Txp.Toil                (Utxo)
+
+----------------------------------------------------------------------------
+-- Initialization
+----------------------------------------------------------------------------
+
+initializeGenesisBalances :: (WithNodeContext ssc m, MonadDB m) => m ()
+initializeGenesisBalances =
+    unlessM hasInitMarker $ do
+        genesisUtxo <- genesisUtxoM
+        putGenesisBalances genesisUtxo
+        putInitMarker
+
+initMarker :: ByteString
+initMarker = "e/init"
```

Comment:
I'm looking at https://github.com/input-output-hk/cardano-sl/blob/2ea6c88d06167c2581a22e3e2e4e3e59d2b9c03c/src/Pos/Txp/DB/Utxo.hs#L212 and it's not there. Should it be added there as well?